### ISSUE TITLE
Tree.Explainer giving different shap values when training xgboost with sparse matrix vs. Pandas DataFrame

### ISSUE THREAD

**Issue by dwoodlock** [shap/shap#984]
If I train an XGBoost model with data created from a Pandas DataFrame, I get very different shap_values than when I train the model with the same data as a sparse matrix.  I don't know why and I don't know which is correct.  I would like sparse to work as I'm using the 'out of core' learning feature of xgboost beta which only works from libsvm files (which behave in the manner that sparse matrices do).  

To recreate, take your Census Income Classification for XGBoost notebook and change your second code cell to:
X, y = shap.datasets.adult()
X_display, y_display = shap.datasets.adult(display=True)

cols = X.columns
X = scipy.sparse.csr_matrix(X.values)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=7)
d_train = xgboost.DMatrix(X_train, label=y_train, feature_names=cols)
d_test = xgboost.DMatrix(X_test, label=y_test, feature_names=cols) 

You'll see the features that are important in the summary plot have changed.  Please advise.  
Don
